File: frontend/src/types/industrialAccident.ts
// Backend: /api/industrial-accidents
import type { PageResponse } from './common'

export type AccidentSeverity = 'MINOR' | 'SERIOUS' | 'FATAL'
export type AccidentType =
  | 'FALL'
  | 'STRUCK'
  | 'CUT'
  | 'BURN'
  | 'ELECTRIC'
  | 'OTHER'
export type VictimGender = 'MALE' | 'FEMALE'

export interface IndustrialAccidentListItem {
  id: number
  accidentNo: string
  companyName: string
  businessNumber: string
  vesselName?: string | null
  accidentDate: string
  accidentLocation: string
  victimName: string
  accidentType: AccidentType
  severity: AccidentSeverity
  createdAt: string
}

export interface IndustrialAccidentDetail extends IndustrialAccidentListItem {
  companyId: number
  vesselId?: number | null
  victimAge: number
  victimGender: VictimGender
  victimRole: string
  description: string
  treatmentDays: number
  absenceDays: number
  reportedBy: number
  reportedByName?: string | null
  reportedAt: string
  reportFileUrl?: string | null
}

export interface IndustrialAccidentPayload {
  companyId: number
  vesselId?: number | null
  accidentDate: string
  accidentLocation: string
  victimName: string
  victimAge: number
  victimGender: VictimGender
  victimRole: string
  accidentType: AccidentType
  severity: AccidentSeverity
  description: string
  treatmentDays: number
  absenceDays: number
}

export interface IndustrialAccidentListParams {
  companyId?: number
  businessNumber?: string
  severity?: AccidentSeverity | ''
  accidentType?: AccidentType | ''
  dateFrom?: string
  dateTo?: string
  page?: number
  size?: number
}

export type IndustrialAccidentPage = PageResponse<IndustrialAccidentListItem>

File: backend/src/main/java/com/penocean/ehs/model/IndustrialAccident.java
package com.penocean.ehs.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class IndustrialAccident {

    private Long id;
    private String accidentNo;              // IA-YYYYMMDD-NNNN
    private Long companyId;
    private String businessNumber;
    private Long vesselId;
    private LocalDate accidentDate;
    private String accidentLocation;
    private String victimName;
    private Integer victimAge;
    private String victimGender;            // M / F
    private String victimRole;
    private String accidentType;            // FALL / STRUCK / CUT / BURN / ELECTRIC / OTHER
    private String severity;                // MINOR / SERIOUS / FATAL
    private String description;
    private Integer treatmentDays;
    private Integer absenceDays;
    private Long reportedBy;
    private LocalDateTime reportedAt;
    private String reportFileUrl;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private Boolean deleted;
}
